source/extensions/filters/network/thrift_proxy/router/router_impl.cc

```diff
@@ -125,6 +126,19 @@ ThriftFilters::FilterStatus Router::messageBegin(MessageMetadataSharedPtr metada
     return ThriftFilters::FilterStatus::StopIteration;
   }
 
+  std::shared_ptr<const ProtocolOptionsConfig> options =
+      cluster_->extensionProtocolOptionsTyped<ProtocolOptionsConfig>(
+          NetworkFilterNames::get().ThriftProxy);
+
+  TransportType transport_type = callbacks_->downstreamTransportType();
+  ProtocolType protocol_type = callbacks_->downstreamProtocolType();
```

Comment:
nit: these could be const, if the conditional from below is folded into these statements with ternary operators